sos/plugins/cs.py

```diff
@@ -1,128 +1,140 @@
+                '/opt/redhat-cs/cert-*/errors',
+                '/opt/redhat-cs/cert-*/system',
+                '/opt/redhat-cs/cert-*/transactions',
+                '/opt/redhat-cs/cert-*/debug',
+                '/opt/redhat-cs/cert-*/tps-debug.log'
             ])
         if csversion == 73:
             self.add_copy_spec([
-                "/var/lib/rhpki-*/conf/*cfg*",
-                "/var/lib/rhpki-*/conf/*.ldif",
-                "/var/lib/rhpki-*/logs/debug",
-                "/var/lib/rhpki-*/logs/catalina.*",
-                "/var/lib/rhpki-*/logs/ra-debug.log",
-                "/var/lib/rhpki-*/logs/transactions",
-                "/var/lib/rhpki-*/logs/system"
-            ])
-        if csversion in (73, 8):
-            self.add_copy_spec([
-                "/etc/dirsrv/slapd-*/dse.ldif",
-                "/var/log/dirsrv/slapd-*/access",
-                "/var/log/dirsrv/slapd-*/errors"
+                '/var/lib/rhpki-*/conf/*cfg*',
+                '/var/lib/rhpki-*/conf/*.ldif',
+                '/var/lib/rhpki-*/logs/debug',
+                '/var/lib/rhpki-*/logs/catalina.*',
+                '/var/lib/rhpki-*/logs/ra-debug.log',
+                '/var/lib/rhpki-*/logs/transactions',
+                '/var/lib/rhpki-*/logs/system'
             ])
         if csversion == 8:
             self.add_copy_spec([
-                "/etc/pki-*/CS.cfg",
-                "/var/lib/pki-*/conf/*cfg*",
-                "/var/log/pki-*/debug",
-                "/var/log/pki-*/catalina.*",
-                "/var/log/pki-*/ra-debug.log",
-                "/var/log/pki-*/transactions",
-                "/var/log/pki-*/system"
+                '/var/lib/pki-*/conf/*cfg*',
+                '/var/log/pki-*/debug',
+                '/var/log/pki-*/catalina.*',
+                '/var/log/pki-*/localhost.*',
+                '/var/log/pki-*/manager.*',
+                '/var/log/pki-*/ra-debug.log',
+                '/var/log/pki-*/transactions',
+                '/var/log/pki-*/selftests.log',
+                '/var/log/pki-*/system'
             ])
         if csversion == 9:
-            # Get logs and configs for each subsystem if installed
-            for subsystem in ('ca', 'kra', 'ocsp', 'tks', 'tps'):
+            for dirs in os.listdir('/var/lib/pki'):
+
+                # Files containing sensitive information.
+                self.add_forbidden_path('/etc/pki/' + dirs
+                                        + '/password.conf')
+                self.add_forbidden_path('/etc/pki/' + dirs
+                                        + '/alias/key3.db')
+
+                # Get certificates from CA.
+                try:
+                    certpath = os.path.join('/var/lib/pki', dirs)
+                    self.add_cmd_output('certutil -L -d %s/alias'
+                                        % certpath)
+                except:
+                    self._log_warn('Could not list /var/lib/pki')
+
```

Comment:
This doesn't work. If you catch an exception in `self.add_cmd_output()` then that is an error *adding* and not *running* the command. It's also extremely unlikely to happen since all we really do in that method is string and list manipulation.

If the intent is to log a warning when the command fails, then there's no need: the command execution code in the `collect()` phase already does that.